### ISSUE TITLE
headline command nodes '@rst-options' and '@rst-option option=value' have no effect

### ISSUE THREAD

**Issue by frakel** [leo-editor/leo-editor#159]
The following nodes ('headline commands')
'@string rst3_stylesheet_name = mycss.css'
or
'@rst-option stylesheet_name = mycss.css'
or
'stylesheet_name = mycss.css' defined in the body section of '@rst-options' 
as described in the Documentation <http://leoeditor.com/rstplugin3.html?#headline-commands>
take no effect on the paramaters handed over to the docutils processor. However, when defined as option doc part (http://leoeditor.com/rstplugin3.html?#option-doc-parts) as part of the @rst node the options are correctly respected in the html result file:

@language rest
@ @rst-options
verbose=True
stylesheet_name=mystyle.css
@c

Best, 
Franz

**Comment by edreamleo** [comment]
Thanks for this report.  It will be fixed asap, and at least by b2.

**Comment by edreamleo** [comment]
**Major code cleaning** Oh dear, I hope I wont regret this ;-)

Rev 262c131 (and following) contain many changes preparing for a new way of handling options for the rst3 and code-to-rst commands. The checkin log::

    Many changes to leoRst.py, preparing to redo how it handles settings.
    Most changes are simply cosmetic.  The real changes:

    - Added top-level new_settings switch.  Now OFF.  All should work as before.  We shall see.
    - rst.getOption now takes a p argument, which tells where to start the scan for arguments.
      This requires pass p to several other methods.
      This is good: we must always tell where to scan for arguments.
    - Added a p argument to all the html parse classes.
      These classes must be created for each node!

All unit tests pass, pylint is happy, and the rst3 command properly generates html/commands.html.
Imo, the old way of handling arguments is essentially impossible to understand and maintain.

The new way will work as follows:

1. A new initSettings method (not written yet) will preprocess the tree, creating self.d, a dict of dicts.  Keys will be vnodes, values will be dicts containing only the @rst option dicts defined in each vnode. Only the top-level command code will call initSettings, which should be a major simplification.

2. getOption will scan p and its parents, looking for the option in d.get(p.v).  If found, we are done, otherwise, the value in a default settings dict, self.d0, will be used.  The code_to_rst uses a scripting dict.  If present, getOption will search it first.

In short, getOption will be the one and only place where "option resolution" will happen.  In effect, the previous code is/was a premature optimization that makes it extremely difficult to init settings properly.

**Comment by edreamleo** [comment]
As discussed [here](https://groups.google.com/d/msg/leo-editor/g6z9P0rPBCs/ELwB89U3KfAJ) we could say that this bug is invalid.  @rst-option/s nodes do work, but they work only for descendants!  This is not likely to be too useful. It may be that not allowing @rst-option and @rst-options in headlines would be best.

In any event, I am thrilled with the new code.  Much simpler, and much faster.

I am going to leave this bug open until the code settles down.  It may be a day or three...

**Comment by edreamleo** [comment]
**New setting**: ```@bool rst3_silent (default False)```

This suppresses the status messages that tell which files have been written.  In practice, these messages aren't very helpful, and they significantly slow the rst3 command.  Coming soon: the rst3 command will always report the *number* of files written.

**Comment by edreamleo** [comment]
The [rst3 reference](http://leoeditor.com/rstplugin3.html#theory-of-operation) now contains reformatted and updated documentation.

**Comment by edreamleo** [comment]
I am going to close this item.  No further work is planned before b1, and besides any work would be an enhancement, not a bug fix.
